driver-core/src/main/com/mongodb/internal/async/AsyncRunnable.java

```diff
@@ -282,4 +284,34 @@ default AsyncRunnable thenRunRetryingWhile(
             ).get(callback);
         });
     }
+
+    /**
+     * This method is equivalent to a do-while loop, where the loop body is executed first and
+     * then the condition is checked to determine whether the loop should continue.
+     *
+     * @param loopBodyRunnable the asynchronous task to be executed in each iteration of the loop
+     * @param whileCheck a condition to check after each iteration; the loop continues as long as this condition returns true
+     * @return the composition of this and the looping branch
+     * @see AsyncCallbackLoop
+     */
+
+    default AsyncRunnable thenRunDoWhileLoop(final AsyncRunnable loopBodyRunnable, final Supplier<Boolean> whileCheck) {
```

Comment:
@vbabanin, you proposed to introduce a type with a static constructing method instead of using `Supplier<Boolean>` so that we could write `while(() -> true)` instead of `() -> true`. And the same can be done for the `AsyncRunnable.thenRunRetryingWhile` method. Did you decide not to do this at all, or do it in a separate PR?

If that improvement is not going to happen in this PR, then at least let's use `BooleanSupplier` instead of `Supplier<Boolean>`.

Reply:
I decided to address the introduction of a specialized type with a static constructing method in a separate PR to avoid overloading this review. For now, I have switched to using BooleanSupplier instead of Supplier<Boolean>, as suggested. Thank you!

Reply:
The change is not in the PR branch.